panels/metrics/cc-metrics-panel.c

```diff
@@ -66,30 +64,26 @@ on_metrics_panel_set_enabled (GObject      *source_object,
     }
 
   self->changing_state = FALSE;
-  self->metrics_active = self->pending_state;
-  gtk_switch_set_state (GTK_SWITCH (self->enable_metrics_switch),
-                        self->metrics_active);
 }
 
 static gboolean
-metrics_switch_active_changed_cb (GtkSwitch *widget,
-                                  gboolean   new_state,
-                                  gpointer   user_data)
+on_metrics_switch_state_set (GtkSwitch *widget,
+                             gboolean   new_state,
+                             gpointer   user_data)
 {
   CcMetricsPanel *self = CC_METRICS_PANEL (user_data);
 
   if (self->changing_state)
     return TRUE;
 
   self->changing_state = TRUE;
```

Comment:
Consider the following:

1. Open this panel
2. Ensure metrics are enabled
3. Send `SIGSTOP` to the metrics daemon so that the SetEnabled call(s) takes a long time
4. Click the switch twice
5. Send `SIGCONT` to the metrics daemon to allow it to reply to method calls

I think the expected behaviour is that clicking twice disables then reënables metrics but by my reading the result will be that metrics are disabled. Is that the desired behaviour? Not sure!

Reply:
I was frowning at that as well but didn't want to mess with it too much. (I think what you describe matches the current behaviour, and other panels like the Location and Camera panels do the same odd thing). I guess we _could_ actually make use of GCancellable here, but I don't think it's a regression.